crates/lib/kajiya-backend/src/vulkan/device.rs

```diff
@@ -601,6 +611,18 @@ impl Device {
     pub fn debug_utils(&self) -> Option<&DebugUtils> {
         self.instance.debug_utils.as_ref()
     }
+
+    pub fn max_bindless_descriptor_count(&self) -> u32 {
+        self.pdevice
+            .properties
+            .limits
+            .max_per_stage_descriptor_sampled_images
+            - RESERVED_DESCRIPTOR_COUNT
```

Comment:
This reports around 4 billion on my device, causing subsequent descriptor set creation to fail.

I have ideas for how to clean it up, but for now let's just do something silly, like:

```suggestion
        (512 * 1024).min(
            self.pdevice
                .properties
                .limits
                .max_per_stage_descriptor_sampled_images
                - RESERVED_DESCRIPTOR_COUNT,
        )
```